src/client/PlayerInfoModal.ts

```diff
@@ -0,0 +1,486 @@
+import { html, LitElement } from "lit";
+import { customElement, query, state } from "lit/decorators.js";
+import { PlayerApiTopSchema, UserMeResponse } from "../core/ApiSchemas";
+import { getServerConfigFromClient } from "../core/configuration/ConfigLoader";
+import { PlayerStats, PlayerStatsSchema } from "../core/StatsSchemas";
+import "./components/baseComponents/PlayerStatsGrid";
+import "./components/baseComponents/PlayerStatsTable";
+
+@customElement("player-info-modal")
+export class PlayerInfoModal extends LitElement {
+  @query("o-modal") private modalEl!: HTMLElement & {
+    open: () => void;
+    close: () => void;
+  };
+
+  @state() private userMeResponse: UserMeResponse | null = null;
+  @state() private visibility: "all" | "public" | "private" = "all";
```

Comment:
1. I don't think we should sum stats from different game types.
2. Please use the existing types from the schema definition for filter. In this case, `GameType`.
```suggestion
  @state() private visibility: GameType = GameType.Public;
```